junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/ClasspathResourceSource.java

```diff
@@ -21,6 +21,7 @@
 import org.junit.platform.commons.util.Preconditions;
 import org.junit.platform.commons.util.ToStringBuilder;
 import org.junit.platform.engine.TestSource;
+import org.junit.platform.engine.discovery.FilePosition;
```

Comment:
This is a breaking change (same above). I'm afraid we'll have to leave the existing `getPosition()` method in place, deprecate it, and add a new `getFilePosition` method that returns an optional of the new type.

Reply:
Good that you caught that one! I've hopefully make the correct changes in https://github.com/junit-team/junit5/pull/2253/commits/a768cf46290e40cc2cd17b2baee630667b2e3b2e ; Could you have a look if this was what you had intended?

Reply:
Also deprecated the old methods in: https://github.com/junit-team/junit5/pull/2253/commits/dd3d8f7a41064a1daadad7b1da9f07b8d67f5170

Reply:
Thanks for the quick turnaround! I'm afraid there's more, though. The old factory methods in the `Source` classes need to be preserved and deprecated, too.

Reply:
Sure thing: https://github.com/junit-team/junit5/pull/2253/commits/f80fcb92f6b60b4400548f2723e0839e0fe145ea

Reply:
Hmm; maybe I was too quick with my change; should this apply to `ClassSource` as well?

Reply:
Figured `ClassSource` probably needed a restore as well, so covered in https://github.com/junit-team/junit5/pull/2253/commits/147ea9d10ebc4b9e6fc9504d230b104c4bd9c70c